plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/CustomServerEvaluationStrategy.java

```diff
@@ -0,0 +1,335 @@
+            globalPropertiesMap.put("externalAddress", externalAddress);
+            globalPropertiesMap.put("externalIP", externalIP);
+            globalPropertiesMap.put("workspaceId", getWorkspaceId());
+            globalPropertiesMap.put("machineName", getMachineName());
+            globalPropertiesMap.put("wildcardNipDomain", getWildcardNipDomain(externalAddress));
+            globalPropertiesMap.put("wildcardXipDomain", getWildcardXipDomain(externalAddress));
+            globalPropertiesMap.put("chePort", chePort);
+        }
+
+        /**
+         * Rendering
+         */
+        public String render(String template, String port) {
+            ST stringTemplate = new ST(template);
+            globalPropertiesMap.forEach((key, value) -> stringTemplate.add(key, value));
+            stringTemplate.add("serverName", portsToRefName.get(port));
+            return stringTemplate.render();
+        }
+
+        /**
+         * Gets the workspace ID from the config of the given container
+         *
+         * @return workspace ID
+         */
+        protected String getWorkspaceId() {
+            return Arrays.stream(env).filter(env -> env.startsWith(CHE_WORKSPACE_ID_PROPERTY))
+                         .map(s -> s.substring(CHE_WORKSPACE_ID_PROPERTY.length()))
+                         .findFirst().get();
+        }
+
+        /**
+         * Gets the workspace Machine Name from the config of the given container
+         *
+         * @return machine name of the workspace
+         */
+        protected String getMachineName() {
+            return Arrays.stream(env).filter(env -> env.startsWith(CHE_MACHINE_NAME_PROPERTY))
+                         .map(s -> s.substring(CHE_MACHINE_NAME_PROPERTY.length()))
+                         .findFirst().get();
+        }
+
+        /**
+         * Gets the IP address of the external address
+         *
+         * @return IP Address
+         */
+        protected String getExternalIp(String externalAddress) {
+            try {
+                return InetAddress.getByName(externalAddress).getHostAddress();
+            } catch (UnknownHostException e) {
+                throw new UnsupportedOperationException("Unable to find the IP for the address '" + externalAddress + "'", e);
+            }
+        }
+
+        /**
+         * Gets a Wildcard domain based on the ip using an external provider nip.io
+         *
+         * @return wildcard domain
+         */
+        protected String getWildcardNipDomain(String externalAddress) {
+            return String.format("%s.%s", getExternalIp(externalAddress), "nip.io");
+        }
+
+        /**
+         * Gets a Wildcard domain based on the ip using an external provider xip.io
+         *
+         * @return wildcard domain
+         */
+        protected String getWildcardXipDomain(String externalAddress) {
+            return String.format("%s.%s", getExternalIp(externalAddress), "xip.io");
```

Comment:
if xip or nip is used, it uses a remote service.
but user can configure its own dns (adding wildcard dns entries) and then it can goes away from xip, nip, etc.